docs/tutorials/timeseriees-forecasting-by-dsp.md

```diff
@@ -1,104 +1,100 @@
+2. The monitored component reports incorrect metric data due to its own bugs.
+3. Application startup consumes far more resources than normal runtime
 
-选取实际序列中所有采样点的$P99.9$和$P0.1$，分别作为上、下限阈值，如果某个采样值低于下限或者高于上限，将采样点的值设置为前一个采样值。
+These extreme anomalies can be disruptive to the signal cycle determination and need to be removed. The approach is as follows:
+
+Select $P99.9$ and $P0.1$ of all sampling points in the actual sequence as the upper and lower thresholds, respectively, and if a sample value is below the lower limit or above the upper limit, set the value of the sampling point to the previous sample value.
 
 ![](../images/algorithm/dsp/remove_outliers.png)
 
-#### 离散傅里叶变换
-对监控的时间序列（设长度为$N$）做快速离散傅里叶变换（FFT），得到信号的频谱图（spectrogram），频谱图直观地表现为在各个离散点$k$处的「冲击」。
-冲击的高度为$k$对应周期分量的「幅度」，$k$的取值范围$\(0,1,2, ... N-1\)$。
+#### Discrete Fourier Transform
+A fast discrete Fourier transform (FFT) is performed on the monitored time series (set length $N$) to obtain a spectrogram of the signal, which is visualized as "shocks" at each discrete point $k$. The height of the shock is the "amplitude" of the corresponding periodic component of $k$, and the value of $k$ ranges from $(0,1,2, ... N-1)$.
 
-$k = 0$对应信号的「直流分量」，对于周期没有影响，因此忽略。
+The "DC component" of the corresponding signal has no effect on the period and is therefore ignored.
 
-由于离散傅里叶变换后的频谱序列前一半和后一半是共轭对称的，反映到频谱图上就是关于轴对称，因此只看前一半$N/2$即可。
+Since the first half and the second half of the discrete Fourier transformed spectral sequence are conjugate symmetric, which is reflected on the spectrogram as being symmetric about the axis, it is sufficient to look at only the first half $$N/2$$.
 
-$k$所对应的周期$$T = {N \over k} \bullet SampleInterval$$
+The corresponding period $$T = {N \over k} \bullet SampleInterval$$
 
-要观察一个信号是不是以$T$为周期，至少需要观察两倍的$T$的长度，因此通过长度为$N$的序列能够识别出的最长周期为$N/2$。所以可以忽略$k = 1$。
+To see if a signal has a period of $$T$$, one needs to observe at least twice the length of $$T$$, so the longest period that can be identified by a sequence of length $$N$$ is $$N/2$$. So $k = 1$ can be ignored.
 
-至此，$k$的取值范围为$(2, 3, ... , N/2)$，对应的周期为$N/2, N/3, ...$，这也就是FFT能够提供的周期信息的「分辨率」。如果一个信号的周期没有落到$N/k$上，它会散布到整个频域，导致「频率泄漏」。
-好在在实际生产环境中，我们通常遇到的应用（尤其是在线业务），如果有规律，都是以「天」为周期的，某些业务可能会有所谓的「周末」效应，即周末和工作日不太一样，如果扩大到「周」的粒度去观察，它们同样具有良好的周期性。
+Up to this point, $k$ takes values in the range $(2, 3, ... , N/2)$, corresponding to the periods $N/2, N/3, ... $, which is the "resolution" of the period information that the FFT can provide. If the period of a signal does not fall on $N/k$, it will scatter to the whole frequency domain, leading to "frequency leakage". The good thing is that in a real production environment, the applications we usually encounter (especially online businesses), if there is a regularity, are on a 'day' cycle, and some businesses may have the so-called 'weekend' effect, i.e., weekends are not quite the same as weekdays, and they are equally well periodic if expanded to the 'week' granularity of observation.
```

Comment:
```suggestion
Up to this point, $k$ takes values in the range $(2, 3, ..., N/2)$, corresponding to the periods $N/2, N/3, ... $, which is the "resolution" of the period information that the FFT can provide. If the period of a signal does not fall on $N/k$, it will scatter to the whole frequency domain, leading to "frequency leakage". The good thing is that in a real production environment, the applications we usually encounter (especially online businesses), if there is regularity, are on a 'day' cycle, and some businesses may have the so-called 'weekend' effect, i.e., weekends are not quite the same as weekdays, and they are equally well periodic if expanded to the 'week' granularity of observation.
```